Tasks/Common/codecoverage-tools/cobertura/cobertura.maven.ccenabler.ts

```diff
@@ -0,0 +1,138 @@
+        let isMultiModule = false;
+        let buildNode = null;
+        
+        if (!buildJsonContent.project) {
+            return Q.reject("Invalid build file");
+        }
+
+        if (buildJsonContent.project.modules) {
+            console.log("Multimodule project detected");
+            isMultiModule = true;
+        }
+
+        if (!buildJsonContent.project.build) {
+            console.log("Build tag is not present");
+            buildJsonContent.project.build = {};
+        }
+
+        if (!buildJsonContent.project.build || typeof buildJsonContent.project.build === "string") {
+            buildNode = {};
+            buildJsonContent.project.build = buildNode;
+        }
+
+        if (buildJsonContent.project.build instanceof Array) {
+            if(typeof buildJsonContent.project.build[0] === 'string'){
+                buildNode = {};
+                buildJsonContent.project.build[0] = buildNode;
+            }else{
+                buildNode = buildJsonContent.project.build[0];
+            }
+        }
+
+        if (buildNode.pluginManagement) {
+            if (typeof buildNode.pluginManagement === "string") {
+                buildNode.pluginManagement = {};
+            }
+            if (buildNode.pluginManagement instanceof Array) {
+                pluginsNode = buildNode.pluginManagement[0].plugins;
+            }
+            else {
+                pluginsNode = buildNode.pluginManagement.plugins;
+            }
+        }
+        else {
+            if (!buildNode.plugins || typeof buildNode.plugins === "string") {
+                buildNode.plugins = {};
+            }
+            if (buildNode.plugins instanceof Array) {
+                if(typeof buildNode.plugins[0] === 'string'){
+                    pluginsNode = {};
+                    buildNode.plugins[0] = pluginsNode;
+                }
+                else{
+                    pluginsNode = buildNode.plugins[0];
+                }
+            }else{
+                pluginsNode = buildNode.plugins;
+            }
+        }
+
+        let ccPluginData = ccc.coberturaMavenEnable(_this.includeFilter, _this.excludeFilter, String(isMultiModule));
+        let reportContent = ccc.coberturaMavenReport();
+
+        return Q.allSettled([ccPluginData, reportContent])
+            .then(function (resp) {
+                util.addPropToJson(pluginsNode, "plugin", resp[0].value.plugin);
+                util.addPropToJson(buildJsonContent.project, "reporting", resp[1].value);
+                console.log("Final buildContent: "+buildJsonContent);
+                return Q.resolve(buildJsonContent);
+            });
+    }
+
+    protected addCodeCoveragePluginData(pomJson: any): Q.Promise<void> {
+        let _this = this;
+        console.log("Adding coverage plugin data");
+        return _this.addCodeCoverageNodes(pomJson)
+            .then(function (buildContent) {
+                return util.writeJsonAsXmlFile(_this.buildFile, buildContent);
```

Comment:
There are examples of using an npm module called xml2js around, I believe both Maven and Gradle are already using it. So you woudn't need to transform to json and back to xml.